src/llmcompressor/modeling/prepare.py

```diff
@@ -35,37 +51,67 @@ def replace_modules_for_calibration(
 
 # ------------------- module replacements; during calibration --------------------
 
-
-def update_qwen3_moe(model, stack, calibrate_all_experts):
-    for module in model.modules():
-        cls_name = module.__class__.__name__
-        if cls_name == "Qwen3MoeDecoderLayer":
-            # Optionally update the model.config to pass in other arguments
-            stack.enter_context(
-                patch_attr(
-                    module,
-                    "mlp",
-                    replace_Qwen3MoE(
-                        config=model.config,
-                        module=module.mlp,
-                        calibrate_all_experts=calibrate_all_experts,
-                    ),
-                )
-            )
+# MoE model configurations - centralized registry
+# Adding a new MoE model is now as simple as adding an entry here!
+# This follows the same pattern as MAPPINGS_REGISTRY in SmoothQuant and AWQ
+MOE_EXPERTS_REPLACEMENTS = {
+    "Qwen3MoeForCausalLM": MoEModelConfig(
+        calibration_type=MoECalibrationType.CONTEXTUAL,
+        target_class_name="Qwen3MoeDecoderLayer",
+        target_attribute="mlp",
```

Comment:
Ideally we shouldn't need to ever target attributes, only target parent modules. For example, only targeting `Qwen3MoeMLP`

Reply:
I'm actually unfamiliar, do we need to specify `target_attribute`s?